magellan-library/src/main/java/com/wealthfront/magellan/core/Navigable.kt

```diff
@@ -1,12 +1,15 @@
 package com.wealthfront.magellan.core
 
+import android.view.Menu
 import android.view.View
 import com.wealthfront.magellan.lifecycle.LifecycleAware
 
 interface Navigable : LifecycleAware {
 
   val view: View?
 
+  fun onUpdateMenu(menu: Menu) {}
```

Comment:
Hmm, do we actually want to do this? Will it propagate it to its children? This doesn't feel like a concern of a Navigable.

The MenuProvider isn't really part of the core library, as it's structured today. So maybe instead of adding this to the `Navigable` interface, we should make a separate interface that Steps can implement when they want to update the menu.

I'm not sure though. Is that too much work? But it also feels weird to add onUpdateMenu to Navigable and not have it as part of the lifecycle or anything.

Reply:
I updated the `MenuComponent` so that it propagates the onUpdateMenu to the navigables, both the journey and the navigable and its children. I think its fine to keep the onUpdateMenu on the Navigable interface for now since all we are saying is that all navigable are gonna be able to control the menu (just like we saw each navigable is gonna be associated with an android view), which i think is valid.

Reply:
That makes sense, but I don't like that it doesn't work out of the box when you attach the expedition to the activity. Imo, we either need to have it completely handled by Magellan (which means we'd probably need to make an official `Expedition` class and make adding more connection points to the `MainActivity` our official connection), or we need to make it completely optional (we don't have an `onUpdateMenu` in the Navigable, and we can make our own API for `MenuProvider`).

Because I would like to avoid creating an actual `Expedition` class, and partially because I think we can come up with a better API than `onUpdateMenu`, I think it should be completely optional. Also, the way this is set up is weirdly close to a lifecycle event without actually using that pattern.

Also, maybe neither of these approaches are correct in the end. We started talking earlier about a way for multiple nested Journeys to control top and bottom chrome without the possibility of it nesting. Maybe this should be a part of that when we build it? If so, I would double down on my opinion that we keep menus out of the core interfaces so it's easier to change later.

Reply:
Hmm. Let me think about this more. For now, I'll just put this on pause. We can come back to this later.